### PR TITLE
CI build for deploy-tool

### PR DESCRIPTION
**Pull request by aaomidi**
Continuous integration for the deploy tool

### CODE REVIEW

File: deploy-tool/cert/kubernetes_storage_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func TestCleanKey(t *testing.T) {
-	prefix := "cm.k8s."
+	prefix := "acme."
```
Comment (tgeoghegan) [inline deploy-tool/cert/kubernetes_storage_test.go:11]: I assume this test was failing without this change? Always good to prove the test works!

File: deploy-tool/.gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 deploy-tool
 config.toml
+vendor/
+coverage.coverprofile
```
Comment (tgeoghegan) [inline deploy-tool/.gitignore:4]: No newline at end of file

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -56,3 +56,35 @@ jobs:
...
+    - name: Test
+      run: go test -race --coverprofile=coverage.coverprofile --covermode=atomic ./...
+      working_directory: deploy-tool
+    
+    - name: Upload test coverage
+      if: success()
+      uses: codecov/codecov-action@v1
+      with:
+        fail_ci_if_error: false
+        token:
```
Comment (tgeoghegan) [inline .github/workflows/ci-build.yml:90]: No newline, and also it seems like we must need to provide a token here? Possibly `${{ secrets.GITHUB_TOKEN }}` is what we want?

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -56,3 +56,35 @@ jobs:
...
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.15
+    
+    - name: Get dependencies
+      run: go get -v -t -d ./...
+      working_directory: deploy-tool
+    
+    - name: Get linter
+      run: go install golang.org/x/lint/golint
```
Comment (tgeoghegan) [inline .github/workflows/ci-build.yml:75]: Do we run this linter after installing it? Is that what `go vet` does?

**Review by tgeoghegan — CHANGES_REQUESTED** [review]
It looks like there are syntax problems with the Github Action that must be resolved.

File: deploy-tool/cert/kubernetes_storage_test.go (hunk 1 above)
Comment (aaomidi) [reply deploy-tool/cert/kubernetes_storage_test.go:11]: Yep!

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (aaomidi) [reply .github/workflows/ci-build.yml:90]: I was looking at a few other places and they...just leave the token blank. Unfortunately I can't test it until its merged it seems :(

File: .github/workflows/ci-build.yml (hunk 4 above)
Comment (aaomidi) [reply .github/workflows/ci-build.yml:75]: ...I don't know how I forgot this.

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (aaomidi) [reply .github/workflows/ci-build.yml:94]: Nevermind I think I can test it.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**